src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -19,17 +20,15 @@ namespace Microsoft.CodeAnalysis.Storage
     /// </summary>
     internal abstract partial class AbstractPersistentStorageService : IChecksummedPersistentStorageService
     {
-        protected readonly IPersistentStorageConfiguration Configuration;
-
         /// <summary>
         /// This lock guards all mutable fields in this type.
         /// </summary>
         private readonly SemaphoreSlim _lock = new(initialCount: 1);
-        private ReferenceCountedDisposable<IChecksummedPersistentStorage>? _currentPersistentStorage;
-        private SolutionId? _currentPersistentStorageSolutionId;
+        private readonly Dictionary<SolutionId, ReferenceCountedDisposable<IChecksummedPersistentStorage>> _solutionIdToStorage = new();
```

Comment:
moved to a model where we can keep alive the storage instance per solution.  the storage service and storage instances are both 'cheap'.  THey do nothing more than call along to their underlying raw-persistence-engines, managing requests  and providing hte necessary data for them.